frontend/packages/operator-lifecycle-manager/src/components/create-operand.tsx

```diff
@@ -104,297 +178,465 @@ const defaultValueFor = (field: OperandField) => {
     return null;
   }
-  if (field.capabilities.includes(SpecCapability.updateStrategy)) {
+
+  // Update strategy
+  if (capabilities.includes(SpecCapability.updateStrategy)) {
     return null;
   }
-  if (field.capabilities.includes(SpecCapability.nodeAffinity)) {
+
+  // Node and pod affinities
+  if (capabilities.includes(SpecCapability.nodeAffinity)) {
     return _.cloneDeep(defaultNodeAffinity);
   }
   if (
-    field.capabilities.includes(SpecCapability.podAffinity) ||
-    field.capabilities.includes(SpecCapability.podAntiAffinity)
+    capabilities.includes(SpecCapability.podAffinity) ||
+    capabilities.includes(SpecCapability.podAntiAffinity)
   ) {
     return _.cloneDeep(defaultPodAffinity);
   }
+
+  // If none of these capabilities are present in the array, return null.
   return null;
 };
 
-const fieldsFor = (providedAPI: CRDDescription) =>
-  _.get(providedAPI, 'specDescriptors', [] as Descriptor[]).map((desc) => ({
-    path: desc.path,
-    displayName: desc.displayName,
-    description: desc.description,
-    type: null,
-    required: false,
-    validation: null,
-    capabilities: desc['x-descriptors'],
-  })) as OperandField[];
-
-const fieldsForOpenAPI = (openAPI: SwaggerDefinition): OperandField[] => {
-  if (_.isEmpty(openAPI)) {
-    return [];
+// Accepts an OpenAPI spec property and returns a corresponding SpecCapability[] array.
+const capabilitiesFor = (property): SpecCapability[] => {
+  if (property.enum) {
+    return property.enum.map((option: string) => `${SpecCapability.select}${option}`);
   }
+  switch (property.type) {
+    case 'integer':
+      return [SpecCapability.number];
+    case 'boolean':
+      return [SpecCapability.booleanSwitch];
+    case 'string':
+    default:
+      return [SpecCapability.text];
+  }
+};
 
-  const fields: OperandField[] = _.flatten(
-    _.map(_.get(openAPI, 'properties.spec.properties', {}), (val, key: string): OperandField[] => {
-      const capabilitiesFor = (property): SpecCapability[] => {
-        if (property.enum) {
-          return property.enum.map((i) => SpecCapability.select.concat(i));
-        }
-        switch (property.type) {
-          case 'integer':
-            return [SpecCapability.number];
-          case 'boolean':
-            return [SpecCapability.booleanSwitch];
-          case 'string':
-          default:
-            return [SpecCapability.text];
-        }
-      };
+/*
+ * Recursively traverses OpenAPI spec properties and flattens all nested properties into an
+ * OperandField[] array.
+ * If a resource instance is provided in the 'obj' argument, then array-type spec properties
+ * will be expanded into the appropriate number of fields.
+ * If providedAPI is included and a field with matching path exists, it's description,
+ * and displayName will take precedence over the corresponding OpenAPI properties, and x-descriptors
+ * will be unioned with the OpenAPI capabilities.
+ */
+const flattenNestedProperties = (
```

Comment:
New helper function which recursively transforms the OpenAPI swagger definitions into a OperandField[] array. This is a modified version of the same function in https://github.com/openshift/console/pull/3366